pkg/kubelet/kubelet.go

```diff
@@ -1759,12 +1760,22 @@ func (kl *Kubelet) syncLoop(updates <-chan kubetypes.PodUpdate, handler SyncHand
 	housekeepingTicker := time.NewTicker(housekeepingPeriod)
 	defer housekeepingTicker.Stop()
 	plegCh := kl.pleg.Watch()
+	const (
+		base   = 100 * time.Millisecond
+		max    = 5 * time.Second
+		factor = 2
+	)
+	duration := base
 	for {
 		if rs := kl.runtimeState.runtimeErrors(); len(rs) != 0 {
 			glog.Infof("skipping pod synchronization - %v", rs)
-			time.Sleep(5 * time.Second)
+			// exponential backoff
```

Comment:
Could [wait.ExponentialBackoff](https://godoc.org/k8s.io/apimachinery/pkg/util/wait#ExponentialBackoff) be used here?

Reply:
I'm not sure about this. Even though you check aggressively here, the other part which updates `runtimeErrors` still works at its original rate, e.g. `updateRuntimeUp` is still called every 5 seconds...

I can imagine this may help in some way, but are we sure this is the right place to put the backoff?

Reply:
This helps the generic case of runtimeErrors. We can certainly add more specific backoffs and/or implement a channel-signaling mechanism in the future to further improve startup performance.